### PR TITLE
NAV-221-clickable-milestones

### PR DESCRIPTION
**Pull request by Manoj-nathwani**
- https://fjelltopp.atlassian.net/browse/NAV-221
- https://youtu.be/EaHXJttKn30

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -51,6 +54,19 @@ export default function IndexPage(props) {
...
   }
+  function updateWorkflowTaskFromMilestoneId(milestoneId) {
+    setloading(true);
+    makeApiRequest(
+      getMilestoneTask(
+        props.currentDatasetId,
+        milestoneId
+      )
+    ).then(({ data }) => {
+      const firstTaskIdInMilestone = data.tasks[0].id;
```
Comment (Manoj-nathwani) [inline pages/index.js:65]: @tomeksabala please confirm there will always be items in this list so that `data.tasks[0].id` would always work

**Comment by jonathansberry** [comment]
I can see from your YouTube vid that you are still highlighting the "future milestone" (the last one) that the user hasn't reached yet as if it is clickable. What happens when you click on that?  I was expecting it to not be clickable for the time being, since we have the more complicated (and blocking) problem of how to show a future task in the `What's Next?` view still to resolve.

File: pages/index.js (hunk 1 above)
Comment (tomeksabala) [reply pages/index.js:65]: I imagine there might be a case to get a milestone with no tasks in it. You should check if tasks.length is non zero first.

**Comment by Manoj-nathwani** [comment]
> I can see from your YouTube vid that you are still highlighting the "future milestone" (the last one) that the user hasn't reached yet as if it is clickable. What happens when you click on that? I was expecting it to not be clickable for the time being, since we have the more complicated (and blocking) problem of how to show a future task in the `What's Next?` view still to resolve.

- @jonathansberry - this is now handled ✅ 
- https://youtu.be/zMabkezOHI4
- Milestones are disabled for all future milestones

**Review by tomeksabala — CHANGES_REQUESTED** [review]
I like the solution. There are a few more issues I think you should fix before it gets merged.
I have converted this PR to a draft as it needs to wait for relevant changes to the api.

**Review by tomeksabala — CHANGES_REQUESTED** [review]
There's a bug in the implementation. Not only future, unreached milestones are disabled for a click but any milestone after the selected one.
So after you click on let's say 2nd milestone you cannot click anymore on milestones 2,3,4...

**Comment by tomeksabala** [comment]
You can test it with real implementation from this PR: https://github.com/fjelltopp/navigator_api/pull/42

**Review by tomeksabala — APPROVED** [review]

**Pull request merged**